amplipi/updater/asgi.py

```diff
@@ -95,6 +98,80 @@ class ReleaseInfo(BaseModel):
+  """
+  conf_path = '/etc/systemd'
+  config = configparser.ConfigParser(strict=False, allow_no_value=True)
+  config.read(f'{conf_path}/journald.conf')
+  return config.get("Journal", "Storage", fallback="") == "persistent"
+
+
+@router.post("/settings/persist_logs")
+def toggle_persist_logs(value: Optional[bool] = None):
+  """
+  Toggles the option within journald to save logs to memory or storage.
+  Accepts an option bool called "value" for if you wish to change the value to a specific option rather than simply flip the state of the config file
+  """
+  try:
+    conf_path = '/etc/systemd'
+    config = configparser.ConfigParser(strict=False, allow_no_value=True)
+    config.read(f'{conf_path}/journald.conf')
+
+    if 'Journal' not in config:
+      config.add_section('Journal')
+
+    # goal_value is a holdover from when this setting was in app.py and toggled via the config settings page
+    # the bool is true if you wish to turn persistent logging on and false if you wish to turn it off
+    # It can be set either with the optional arg of the function or it attempts to find the current value from the config file and set the opposite of that
+    goal_value: bool
+    if value is not None:
+      goal_value = value
+    else:
+      goal_value = not get_log_persist_state()  # goal_value is opposite of present value as a default/fallback
+
+    if goal_value:  # Set persist
+      config.set('Journal', 'Storage', 'persistent')
+      config.set('Journal', 'SyncIntervalSec', '30s')
+      config.set('Journal', 'SystemMaxUse', '64M')
+
+      config.remove_option('Journal', 'RuntimeMaxUse')
+      config.remove_option('Journal', 'ForwardToConsole')
+      config.remove_option('Journal', 'ForwardToWall')
+
+      logger.info("persist_logs activated!")
+
+    else:  # Reset config to default as seen in configure.py
+      config.set('Journal', 'Storage', 'volatile')
+      config.set('Journal', 'RuntimeMaxUse', '64M')
+      config.set('Journal', 'ForwardToConsole', 'no')
+      config.set('Journal', 'ForwardToWall', 'no')
+
+      config.remove_option('Journal', 'SyncIntervalSec')
+      config.remove_option('Journal', 'SystemMaxUse')
+
+      logger.info("persist_logs deactivated!")
+
+    with open('/tmp/journald.conf.tmp', 'w', encoding="utf-8") as conf_file:
+      config.write(conf_file)
+
+    subprocess.run(['sudo', 'mv', '/tmp/journald.conf.tmp', f'{conf_path}/journald.conf'], check=True)
+    subprocess.run(['sudo', 'systemctl', 'restart', 'systemd-journald'], check=True)
+  except Exception as exc:
+    logger.error("persist_logs toggle failed!")
+    raise exc
+
+
+@router.delete("/settings/persist_logs")
+def delete_logs():
+  """Deletes everything in the persistent logs directory if logs are not set to persist"""
+  if not get_log_persist_state():
+    subprocess.run(['sudo', 'rm', '/var/logs/journal/*'], check=True)
```

Comment:
Nice and simple deletion endpoint, currently only called by a curl command inside of a bash file that's called by a cronjob at midnight

Reply:
considering I don't want to remove logs via cronjob, is this necessary? I would argue not. It also feels like a somewhat privileged action - an arbitrary user on the network should not be able to hide evidence of malfeasance by hitting an unauthenticated API endpoint.